### PR TITLE
Store migrations

### PR DESCRIPTION
**Pull request by viren-nadkarni**
### Summary

Refactors these services from RegionBackend to Stores.

- CloudFormation
- API Gateway v1
- OpenSearch
- Kinesis
- Route 53
- SQS (Also reorganises the code into submodules (`.constants`, `.utils`, `.models`, `.exceptions`) cc: @thrau 
- DynamoDB (Previously two stores were defined. Now this is consolidated. Going forward we want to keep one store only as it is simpler and avoid patching) cc: @giograno 
- DynamoDB Streams
- Firehose
- EventBridge
- SNS

This change will introduce full multi-accounts + persistence + cloud pods support to all community providers.

### CODE REVIEW

File: localstack/services/cloudformation/models/__init__.py [hunk 1]
```diff
@@ -25,4 +25,5 @@
     "sqs",
     "ssm",
     "stepfunctions",
+    "stores",
```
Comment (whummer) [inline localstack/services/cloudformation/models/__init__.py:28]: CloudFormation is a bit of a special case - the `models` package is more intended for the CFn resource model implementations (e.g., `AWS::Lambda::Function`, etc). 

I understand that the name clashes with the general convention of putting stores into `models.py` - but perhaps we could move the file to `services/cloudformation/stores.py`?

File: tests/integration/test_sns.py [hunk 2]
```diff
@@ -467,7 +468,7 @@ def test_subscribe_platform_endpoint(
         self, sns_client, sqs_create_queue, sns_create_topic, sns_subscription
     ):
 
-        sns_backend = SNSBackend.get()
+        sns_backend = sns_stores[get_aws_account_id()][aws_stack.get_region()]
```
Comment (whummer) [inline tests/integration/test_sns.py:471]: nit: I guess we could simply use `SnsProvider.get_store()` here, which would be slightly shorter

**Review by whummer — APPROVED** [review]
Awesome set of changes @viren-nadkarni 🚀 - kudos, again, for churning through all the backends and making the switch to Stores! Excited to see it in action with the pods integration soon 🎉

File: localstack/services/cloudformation/models/__init__.py (hunk 1 above)
Comment (viren-nadkarni) [reply localstack/services/cloudformation/models/__init__.py:28]: Thanks, I wasn't sure if the old path was appropriate.

Fixed in 36957feb

File: tests/integration/test_sns.py (hunk 2 above)
Comment (viren-nadkarni) [reply tests/integration/test_sns.py:471]: Good catch, fixed with 36957feb

**Pull request merged**